test/visibility/import/privateModuleMethodAccess.chpl

```diff
@@ -8,6 +8,7 @@ module A {
     const c: bool;
   }
   private module Inner {
+    import A.Foo;
     proc Foo.hidden() {
```

Comment:
How do you feel about an additional version of this test that has `import A; proc A.Foo.hidden()`?

Reply:
I don't think that's supported even on master (?).  On master, if I say `proc A.foo.hidden()` I get a syntax error.  If I say `proc (A.foo).hidden()` then I get a (potentially buggy?) resolution error message:

```
> privateModuleMethodAccess2.chpl:17: In function 'callHidden':
> privateModuleMethodAccess2.chpl:18: error: unresolved call 'hidden()'
> privateModuleMethodAccess2.chpl:12: note: this candidate did not match: (A.Foo).hidden()
> privateModuleMethodAccess2.chpl:18: note: because call is not written as a method call
> privateModuleMethodAccess2.chpl:12: note: but candidate function is a method
> privateModuleMethodAccess2.chpl:27: Function 'callHidden' instantiated as: callHidden(this: borrowed Foo(6,real(64)))
```

I don't think my PR changed anything about this behavior.  Tagging @mppf since he's most familiar with the parenthesized target form of writing secondary methods and implemented these explanation-based resolution errors.

Reply:
> I don't think my PR changed anything about this behavior.

I don't think so either.  I'm more thinking through what it would be reasonable for a user to write when adjusting their code because of this change.  This means that submodules need to explicitly bring in classes or records that were otherwise available.  Just something to consider when weighing the impact of this change

Reply:
> This means that submodules need to explicitly bring in classes or records that were otherwise available.

I don't understand...  under https://github.com/Cray/chapel-private/issues/884, submodules need to explicitly bring in everything in their parent module's scope that they were relying on lexical scoping for before, classes, records, or whatever.  They could write `use A;` rather than `import A.Foo;` if they didn't want to name the class directly.

Reply:
> submodules need to explicitly bring in everything in their parent module's scope that they were relying on lexical scoping for before, classes, records, or whatever.

Right.  This is as much for me making sure I understand the impact, and whether this should be consistent or whether it implies we're going to have more trouble for the secondary method case again

Reply:
OK.  So is there more you're asking me to do here for this PR?

Regarding secondary methods, I wrote up the following case, and was happy to see that it behaved as I expected:

```chapel
module M {
  use Child;

  proc C.foo() {
    writeln("In C.foo()");
  }

  module Child {
    class C {
    }
  }

  module Child2	{
    proc main() {
      import M.Child.C;  // import class C, but nothing in my parent module
      var myC = new C();
      myC.foo();
    }
  }
}
```

Reply:
(offline, Lydia requested that I file this test and call it good at that).

Reply:
>  Tagging @mppf since he's most familiar with the parenthesized target form of writing secondary methods and implemented these explanation-based resolution errors.

I don't think I've ever tried to write `proc (Module.MyType).method()` before so it's not really surprising to me it didn't work. Could make a good issue/future...